### PR TITLE
SearchAPI

### PR DESCRIPTION
**Pull request by RanjithTom**
* Updated query param pageSize into results
* added new query param page to indicate from where results should be returned

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 1]
```diff
@@ -45,6 +45,7 @@ public class ElasticsearchTest {
     public static final long DELAY_AFTER_INDEXING = 1000L;
     private static final String ELASTICSEARCH_VERSION = "7.10.2";
     private static final int RESULT_SIZE = 10;
+    private static final int RESULT_FROM = 0;
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:48]: Better FROM_BEGINNING

File: search-resources-api/src/test/java/no/unit/nva/search/SearchAllHandlerTest.java [hunk 2]
```diff
@@ -3,7 +3,9 @@
 import static java.net.HttpURLConnection.HTTP_OK;
 import static no.unit.nva.search.RequestUtil.DOMAIN_NAME;
 import static no.unit.nva.search.RequestUtil.PATH;
-import static no.unit.nva.search.SearchAllHandler.PAGE_SIZE_QUERY_PARAM;
+import static no.unit.nva.search.SearchAllHandler.DEFAULT_RESULTS_SIZE;
+import static no.unit.nva.search.SearchAllHandler.PAGE_QUERY_PARAM;
+import static no.unit.nva.search.SearchAllHandler.RESULTS_QUERY_PARAM;
 import static no.unit.nva.search.SearchHandler.DEFAULT_PAGE_SIZE;
 import static no.unit.nva.search.SearchHandler.EXPECTED_ACCESS_RIGHT_FOR_VIEWING_MESSAGES_AND_DOI_REQUESTS;
 import static no.unit.nva.search.SearchHandler.VIEWING_SCOPE_QUERY_PARAMETER;
```
Comment (axthosarouris) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchAllHandlerTest.java:11]: we have a repetition of constants here. Better to move the constants in its own class

File: search-resources-api/src/main/java/no/unit/nva/search/SearchAllHandler.java [hunk 3]
```diff
@@ -71,10 +74,20 @@ private static IdentityClient defaultIdentityClient() {
...
+    }
+
+    private Integer extractPageNo(RequestInfo requestInfo) {
+        return requestInfo.getQueryParameterOpt(PAGE_QUERY_PARAM)
+            .map(Integer::valueOf)
+            .orElse(DEFAULT_RESULTS_INDEX);
+    }
+
+    private Integer calculateResultsFromIndex(RequestInfo requestInfo) {
+        return extractPageNo(requestInfo) * extractResultsSize(requestInfo);
```
Comment (axthosarouris) [inline search-resources-api/src/main/java/no/unit/nva/search/SearchAllHandler.java:90]: Each class should have only one responsibility (Single-responsibility principle). The responsibility of this class is to get the REST request and forward the input parameters to the  associated services. The calculations should happen in the SearchClient and not here. Then you can reuse the code for the SearchHandler that is used for searching publications. SearchAll handler searches DoiRequests and SupportMessages. The class names are a bit wrong

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 4]
```diff
@@ -102,27 +94,34 @@ public SearchResponse search(SearchRequest searchRequest, RequestOptions request
...
     }
 
-    private ViewingScope getSampleViewingScope() {
-        ViewingScope viewingScope = new ViewingScope();
-        viewingScope.setIncludedUnits(Set.of(randomUri(), randomUri()));
-        viewingScope.setExcludedUnits(Set.of(randomUri()));
-        return viewingScope;
-    }
+    @Test
+    void shouldReturnResponseWithResultsFromDefinedFromWhenSearchingForResources() throws ApiGatewayException {
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:105]: This test name and the the test name of the test before is misleading. We test only what is sent to ES, not what ES is replying to us.

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java (hunk 1 above)
Comment (RanjithTom) [reply search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:48]: What about BEGINNING_FROM 🤔 , better readability

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java [hunk 5]
```diff
@@ -48,11 +48,12 @@ public SearchResourcesResponse searchSingleTerm(SearchDocumentsQuery query, Stri
 
     public SearchResponse findResourcesForOrganizationIds(ViewingScope viewingScope,
                                                           int resultSize,
-                                                          String... index)
+                                                          int pageNo, String... index)
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/SearchClient.java:51]: Each parameter in single line.

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java [hunk 6]
```diff
@@ -48,11 +48,12 @@ public SearchResourcesResponse searchSingleTerm(SearchDocumentsQuery query, Stri
 
     public SearchResponse findResourcesForOrganizationIds(ViewingScope viewingScope,
                                                           int resultSize,
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/SearchClient.java:50]: I think the resultSize should be called internally pageSize. It makes more sense

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java [hunk 7]
```diff
@@ -70,15 +71,20 @@ public SearchResponse doSearch(SearchDocumentsQuery query, String index) throws
...
 
     private SearchRequest createSearchRequestForResourcesWithOrganizationIds(
-        ViewingScope viewingScope, int resultSize, String... indices) {
+        ViewingScope viewingScope, int resultSize, int pageNo, String... indices) {
         BoolQueryBuilder queryBuilder = matchOneOfOrganizationIdsQuery(viewingScope);
         SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
             .query(queryBuilder)
-            .size(resultSize);
+            .size(resultSize)
+            .from(calculateResultsFrom(resultSize, pageNo));
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/SearchClient.java:79]: calculateFirstEntryIndex

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 8]
```diff
@@ -45,6 +45,7 @@ public class ElasticsearchTest {
     public static final long DELAY_AFTER_INDEXING = 1000L;
     private static final String ELASTICSEARCH_VERSION = "7.10.2";
     private static final int RESULT_SIZE = 10;
+    private static final int BEGINNING_FROM = 0;
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:48]: I think this should be called PAGE_NUMBER?

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 9]
```diff
@@ -80,7 +74,7 @@ void shouldReturnSearchResponseWhenSearchingWithOrganizationIds() throws ApiGate
             new SearchClient(new RestHighLevelClientWrapper(restHighLevelClient));
         SearchResponse response =
             searchClient.findResourcesForOrganizationIds(getSampleViewingScope(),
-                                                         10,
+                                                         DEFAULT_RESULT_SIZE, 0,
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:77]: replace 0 with constant

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 10]
```diff
@@ -102,27 +95,35 @@ public SearchResponse search(SearchRequest searchRequest, RequestOptions request
...
     }
 
-    private ViewingScope getSampleViewingScope() {
-        ViewingScope viewingScope = new ViewingScope();
-        viewingScope.setIncludedUnits(Set.of(randomUri(), randomUri()));
-        viewingScope.setExcludedUnits(Set.of(randomUri()));
-        return viewingScope;
-    }
+    @Test
+    void shouldSentRequestWithDefinedFromWhenSearchingForResourcesWithPageNo() throws ApiGatewayException {
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:106]: shouldSendRequestWithFirstEntryIndexCalcuatedBySuppliedPageSizeAndPageNumber

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 11]
```diff
@@ -158,6 +159,29 @@ void searchSingleTermReturnsErrorResponseWhenExceptionInDoSearch() throws IOExce
                      () -> searchClient.searchSingleTerm(getSampleQuery(), ELASTICSEARCH_ENDPOINT_INDEX));
     }
 
+    private static URI getSampleRequestUri() {
+        return new UriWrapper("https", "localhost")
+            .addChild("search")
+            .addChild("resources")
+            .getUri();
+    }
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:167]: remove this and use the method `randomUri()`

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 12]
```diff
@@ -158,6 +159,29 @@ void searchSingleTermReturnsErrorResponseWhenExceptionInDoSearch() throws IOExce
...
+
+    private ViewingScope getSampleViewingScope() {
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:169]: do not use get for things that are not getters

File: search-resources-api/src/test/java/no/unit/nva/search/SearchAllHandlerTest.java [hunk 13]
```diff
@@ -175,7 +205,7 @@ void shouldSendQueryWithViewingScopeEqualToUserTopCristingOrgIdWhenNoOtherViewin
     }
 
     @Test
-    void shouldSentRequestWithGivenPageSize() throws IOException {
+    void shouldSentRequestWithGivenResults() throws IOException {
```
Comment (axthosarouris) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchAllHandlerTest.java:208]: shouldSend

**Review by axthosarouris — CHANGES_REQUESTED** [review]
General comments: 
1. Pay attention to formatting. Breaking formatting means sloppy code.
2. Fix test names to represent what we are actually testing.

**Pull request merged**